x-pack/plugins/rule_registry/server/utils/create_lifecycle_executor.ts

```diff
@@ -74,7 +74,7 @@ export interface LifecycleAlertServices<
   alertWithLifecycle: LifecycleAlertService<InstanceState, InstanceContext, ActionGroupIds>;
   getAlertStartedDate: (alertInstanceId: string) => string | null;
   getAlertUuid: (alertInstanceId: string) => string;
-  getAlertByAlertUuid: (alertUuid: string) => { [x: string]: any } | null;
+  getAlertByAlertUuid: (alertUuid: string) => Promise<Array<{ [id: string]: any }> | null>;
```

Comment:
A bit confused on this.  The typing in the interface and mock for `getAlertByAlertUuid()` return value changed - pretty significantly - but the actual implementation (also in this file, near line 200) did not?  Seems like the types are not working correctly here.  Or maybe I'm missing something ...

Reply:
The types were originally defined incorrectly, `getAlertByAlertUuid` implementation returned a promise that was either an array or null. The mock was returning the incorrect values as well.

Reply:
Cool, seemed like something like that.

I guess I'm wondering how the types failed us before (not noticing the difference), and if that part is fixed.

Reply:
The signature of `getAlertByAlertUuid` should be the same as `fetchAlertByUuid` which looks like it's just returning `hits` prop from the response from `ruleDataClient.getReader().search(request)` which returns the results from this interface: 

```
  search<TSearchRequest extends ESSearchRequest>(
    request: TSearchRequest
  ): Promise<
    ESSearchResponse<Partial<ParsedTechnicalFields & ParsedExperimentalFields>, TSearchRequest>
  >;
```
which ends up being a type that's made up by a bunch of generics (which probably intersects with `Array<{ [x: string]: any }>` at some point). I would argue that `fetchAlertByUuid` should return an actual type for an `Alert` since the name of the function implies that it's returning a single alert and not an array of hits;  which is made up by a complicated generic (of generics).

Getting into a deep Typescript refactoring exercise is outside the scope of this PR. This is something the Response Ops team should keep in mind as they work on making the Rule Registry more ubiquitous.

Reply:
Gotcha, thx - figured it was something like that ...